src/ios/Flic.m

```diff
@@ -0,0 +1,175 @@
+//
+//  Flic
+//
+//  Created by Maxim Dukhanov <m.dukhanov@gmail.com>
+//
+
+#import "Flic.h"
+#import <Cordova/CDVAvailability.h>
+#import <fliclib/fliclib.h>
+
+
+@interface Flic () <SCLFlicManagerDelegate, SCLFlicButtonDelegate>
+@end
+
+@implementation Flic
+
+static NSString * const pluginNotInitializedMessage = @"flic is not initialized";
+static NSString * const TAG = @"[TAF Flic] ";
+@synthesize onButtonClickCallbackId;
+
+- (void)pluginInitialize
+{
+    [self log:@"pluginInitialize"];
+    //[[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(openApplicationURL:) name:UIApplicationLaunchOptionsURLKey object:nil];
+}
+
+- (BOOL)openApplicationURL:(NSNotification *)notification
+{
+    [self log:@"openApplicationURL"];
+    BOOL wasHandled = NO;
+    //wasHandled = [[SCLFlicManager sharedManager] handleOpenURL:url];
+    return wasHandled;
+}
+
+- (void) init:(CDVInvokedUrlCommand*)command
+{
+    [self log:@"init"];
+    
+	NSDictionary *config = [command.arguments objectAtIndex:0];
+	NSString* APP_ID = [config objectForKey:@"appId"];
+	NSString* APP_SECRET = [config objectForKey:@"appSecret"];
+	
+    self.flicManager = [SCLFlicManager configureWithDelegate:self defaultButtonDelegate:self appID:APP_ID appSecret:APP_SECRET backgroundExecution:NO];
+    
+    CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:[self knownButtons]];
+    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId ];
+}
+
+- (void) getKnownButtons:(CDVInvokedUrlCommand*)command
+{
+    [self log:@"getKnownButtons"];
+    
+    CDVPluginResult* result;
+    // in case plugin is not initialized
+    if (self.flicManager == nil) {
+        result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:pluginNotInitializedMessage];
+        [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
+        return;
+    }
+    
+	result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:[self knownButtons]];
+	[self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
+}
+
+- (void) grabButton:(CDVInvokedUrlCommand*)command
+{
+    [self log:@"grabButton"];
+    
+    CDVPluginResult* result;
+    // in case plugin is not initialized
+    if (self.flicManager == nil) {
+        [self log:@"flicManager is null"];
+        result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:pluginNotInitializedMessage];
+        [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
+        return;
+    }
+    
+	[[SCLFlicManager sharedManager] grabFlicFromFlicAppWithCallbackUrlScheme:@"winfleet-tracker"];
+	result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+	[self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
+}
+
```

Comment:
I recently merged a pull request from another user who proposed changes to the JavaScript interface, such as eliminating the waitForButtonEvent and triggerButtonEvent. Now on Android it works in push mode. I guess I can delete these methods and the declaration in the .h file. Right?